crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -563,24 +565,22 @@ where
                 }
             }
             ast::Stmt::Assign(node) => {
-                debug_assert!(self.current_assignment.is_none());
```

Comment:
I think it's still correct to assert that the length of `self.current_assignments` is 0 here? A NamedExpr should be the only kind of assignment that can be nested inside the target of another assignment.

Same for all other cases, except for `NamedExpr`, where this assertion was removed.

Reply:
That's a good point, since you don't have statements in statements. Re-evaluated the changeset and agree with the assertion that we can keep all the assertions.